homeassistant/components/mobile_app/__init__.py

```diff
@@ -1,355 +1,62 @@
-})
-
-UPDATE_DEVICE_SCHEMA = vol.Schema({
-    vol.Optional(ATTR_APP_DATA, default={}): dict,
-    vol.Required(ATTR_APP_VERSION): cv.string,
-    vol.Required(ATTR_DEVICE_NAME): cv.string,
-    vol.Required(ATTR_MANUFACTURER): cv.string,
-    vol.Required(ATTR_MODEL): cv.string,
-    vol.Optional(ATTR_OS_VERSION): cv.string,
-})
-
-WEBHOOK_PAYLOAD_SCHEMA = vol.Schema({
-    vol.Required(ATTR_WEBHOOK_TYPE): vol.In(WEBHOOK_TYPES),
-    vol.Required(ATTR_WEBHOOK_DATA, default={}): dict,
-    vol.Optional(ATTR_WEBHOOK_ENCRYPTED, default=False): cv.boolean,
-    vol.Optional(ATTR_WEBHOOK_ENCRYPTED_DATA): cv.string,
-})
-
-CALL_SERVICE_SCHEMA = vol.Schema({
-    vol.Required(ATTR_DOMAIN): cv.string,
-    vol.Required(ATTR_SERVICE): cv.string,
-    vol.Optional(ATTR_SERVICE_DATA, default={}): dict,
-})
-
-FIRE_EVENT_SCHEMA = vol.Schema({
-    vol.Required(ATTR_EVENT_TYPE): cv.string,
-    vol.Optional(ATTR_EVENT_DATA, default={}): dict,
-})
-
-RENDER_TEMPLATE_SCHEMA = vol.Schema({
-    vol.Required(ATTR_TEMPLATE): cv.string,
-    vol.Optional(ATTR_TEMPLATE_VARIABLES, default={}): dict,
-})
-
-WEBHOOK_SCHEMAS = {
-    WEBHOOK_TYPE_CALL_SERVICE: CALL_SERVICE_SCHEMA,
-    WEBHOOK_TYPE_FIRE_EVENT: FIRE_EVENT_SCHEMA,
-    WEBHOOK_TYPE_RENDER_TEMPLATE: RENDER_TEMPLATE_SCHEMA,
-    WEBHOOK_TYPE_UPDATE_LOCATION: SEE_SCHEMA,
-    WEBHOOK_TYPE_UPDATE_REGISTRATION: UPDATE_DEVICE_SCHEMA,
-}
-
-
-def get_cipher():
-    """Return decryption function and length of key.
-
-    Async friendly.
-    """
-    from nacl.secret import SecretBox
-    from nacl.encoding import Base64Encoder
-
-    def decrypt(ciphertext, key):
-        """Decrypt ciphertext using key."""
-        return SecretBox(key).decrypt(ciphertext, encoder=Base64Encoder)
-    return (SecretBox.KEY_SIZE, decrypt)
-
-
-def _decrypt_payload(key, ciphertext):
-    """Decrypt encrypted payload."""
-    try:
-        keylen, decrypt = get_cipher()
-    except OSError:
-        _LOGGER.warning(
-            "Ignoring encrypted payload because libsodium not installed")
-        return None
-
-    if key is None:
-        _LOGGER.warning(
-            "Ignoring encrypted payload because no decryption key known")
-        return None
-
-    key = key.encode("utf-8")
-    key = key[:keylen]
-    key = key.ljust(keylen, b'\0')
-
-    try:
-        message = decrypt(ciphertext, key)
-        message = json.loads(message.decode("utf-8"))
-        _LOGGER.debug("Successfully decrypted mobile_app payload")
-        return message
-    except ValueError:
-        _LOGGER.warning("Ignoring encrypted payload because unable to decrypt")
-        return None
-
-
-def context(device):
-    """Generate a context from a request."""
-    return Context(user_id=device[CONF_USER_ID])
+from .http_api import register_http_handlers
+from .webhook import setup_device
+from .websocket_api import register_websocket_handlers
 
+DEPENDENCIES = ['device_tracker', 'http', 'webhook', 'websocket_api']
```

Comment:
Drop the websocket_api dependency. Websocket commands can register without, and it's only needed for configuration.